mesonbuild/interpreterbase/interpreterbase.py

```diff
@@ -54,6 +54,8 @@
 if T.TYPE_CHECKING:
     from ..interpreter import Interpreter
 
+    SubProject = T.NewType('SubProject', str)
```

Comment:
You gnerallly don't put these inside a TYPE_CHECKING block, because you can do `SubProject('subproject_name')` otherwise.

Reply:
Interesting, I didn't realize NewType returned something useful outside of type checking. Apparently `X(y)` does essentially the same as `T.cast(X, y)`, although with the technical downside of actually evaluating a `T.NewType.__init__()` at runtime so you can use it at runtime.

Even as a micro-optimization it's debatable if that is worth caring about. Might as well do as you say.